src/flash-source-buffer.js

```diff
@@ -304,47 +304,68 @@ export default class FlashSourceBuffer extends videojs.EventTarget {
   convertTagsToData_(segmentData) {
     let segmentByteLength = 0;
     let tech = this.mediaSource_.tech_;
-    let targetPts = 0;
+    let targetVideoPts = 0;
+    let targetAudioPts = 0;
     let i;
     let j;
     let segment;
-    let filteredTags = [];
-    let tags = this.getOrderedTags_(segmentData);
+    let filteredAudioTags = [];
+    let filteredVideoTags = [];
+    let videoTags = segmentData.tags.videoTags;
+    let audioTags = segmentData.tags.audioTags;
 
     // Establish the media timeline to PTS translation if we don't
     // have one already
-    if (isNaN(this.basePtsOffset_) && tags.length) {
-      this.basePtsOffset_ = tags[0].pts;
+    if (isNaN(this.basePtsOffset_) && (videoTags.length || audioTags.length)) {
+      const firstVideoTag = videoTags[0] || { pts: Infinity };
+      const firstAudioTag = audioTags[0] || { pts: Infinity };
+
+      this.basePtsOffset_ = Math.min(firstAudioTag.pts, firstVideoTag.pts);
+    }
+
+    if (tech.buffered().length) {
+      targetAudioPts = tech.buffered().end(0) - this.timestampOffset;
     }
 
     // Trim to currentTime if it's ahead of buffered or buffered doesn't exist
     if (tech.seeking()) {
-      targetPts = Math.max(targetPts, tech.currentTime() - this.timestampOffset);
+      targetVideoPts = Math.max(targetVideoPts, tech.currentTime() - this.timestampOffset);
+      targetAudioPts = Math.max(targetAudioPts, tech.currentTime() - this.timestampOffset);
     }
 
     // PTS values are represented in milliseconds
-    targetPts *= 1e3;
-    targetPts += this.basePtsOffset_;
+    targetVideoPts *= 1e3;
+    targetVideoPts += this.basePtsOffset_;
+    targetAudioPts *= 1e3;
+    targetAudioPts += this.basePtsOffset_;
 
     // skip tags with a presentation time less than the seek target
-    for (i = 0; i < tags.length; i++) {
-      if (tags[i].pts >= targetPts) {
-        filteredTags.push(tags[i]);
+    for (i = 0; i < videoTags.length; i++) {
+      if (videoTags[i].pts >= targetVideoPts) {
+        filteredVideoTags.push(videoTags[i]);
```

Comment:
We seem to not culling any video tags at all (effectively). I think the idea should cull with a bit of awareness of the GOP structure and cull entire GOPS that precede the `targetVideoPts` (which would be the seek time or the end of the buffer). Doing this means playback starts within complete GOPS and should eliminate the "smearing" that was experienced previously without pushing too many extra frames into Flash.